pkg/controller/utils/auth.go

```diff
@@ -0,0 +1,98 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-ldap/ldap"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/render"
+	rauth "github.com/tigera/operator/pkg/render/common/authentication"
+	"github.com/tigera/operator/pkg/render/common/authentication/tigera/key_validator_config"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// GetKeyValidatorConfig uses the operatorv1.Authentication CR given to create the KeyValidatorConfig. This may be
+// either a DexKeyValidatorConfig or a tigerakvc.KeyValidatorConfig.
+func GetKeyValidatorConfig(ctx context.Context, cli client.Client, authenticationCR *operatorv1.Authentication, clusterDomain string) (rauth.KeyValidatorConfig, error) {
+	var keyValidatorConfig rauth.KeyValidatorConfig
+	if authenticationCR != nil {
+		idpSecret, err := GetIdpSecret(ctx, cli, authenticationCR)
+		if err != nil {
+			return nil, err
+		}
+
+		oidc := authenticationCR.Spec.OIDC
+		if authenticationCR.Spec.OIDC != nil && authenticationCR.Spec.OIDC.Type == operatorv1.OIDCTypeTigera {
```

Comment:
You created this nice alias but then you hardly used it!
```suggestion
		if oidc != nil && oidc.Type == operatorv1.OIDCTypeTigera {
```